### ISSUE TITLE
fullscreen global: black screen after update to 1.10

### ISSUE THREAD

**Issue by 532910** [swaywm/sway#8419]
after update to 1.10 (debian-experimental) fullscreen global just shows black screen, even with only one monitor

**Comment by rod-stuchi** [comment]
In my case (running on Arch, 6.11.5-arch1-1), Sway 1.10

 when I try to put some application on fullscreen `$mod+f` the mouse cursor locks and nothing else happens

**Comment by rod-stuchi** [comment]
global `fullscreen` is working for me, only problem with simple `fullscreen` 

```
bindsym --to-code $mod+f fullscreen global
bindsym --to-code $mod+Shift+f fullscreen
```

**Comment by 532910** [comment]
> when I try to put some application on fullscreen $mod+f the mouse cursor locks and nothing else happens

> bindsym --to-code $mod+f fullscreen global

so mod+f is a fullscreen global in your case

**Comment by rod-stuchi** [comment]
> > when I try to put some application on fullscreen $mod+f the mouse cursor locks and nothing else happens
> 
> > bindsym --to-code $mod+f fullscreen global
> 
> so mod+f is a fullscreen global in your case

Nope, I just copied your keybindings and changed to `f` instead of `a` to test.

tried with `swaymsg 'fullscreen enable' ` in a terminal (wezterm), same issue

**Comment by egnappahz** [comment]
I have the same problem. I got 3 screens so global full screen is not a valid workaround for me. I fullscreen per monitor(workspace).
I got more additional confusing info!
On some specific screens when applications go fullscreen by accident (when it spans accross the entire screenarea), that application freezes aswel. These weird freezes happens on 2 of my 3 screens, not my "main" one, although I dont think wayland even has the concept "main screen". When I am able to resize/move the application so they dont span across the entire area of the screen, the freezing of the app stops.
But I must admit, I have different GPUs driving different monitors, so I have too many moving parts here to give a clear assesment.

**Comment by benjamineeckh** [comment]
Same problem over here (black background), nothing changed just upgraded to v1.10. It also does not respect my window layout anymore, they are suddenly swapped (ie. 1|2 became 2|1).
I am also using "fullscreen" for the fullscreen toggle. (bindsym $mod+f fullscreen)

Also locks up graphically when fullscreening a window. But **only** graphically, my system is still responsive, just the display where the pdf was openened does not update anymore.
I can also just focus the display and close the pdf/un-fullscreen it, after which the glitch is gone.

Running on 6.11.5-arch1-1

**Comment by Nefsen402** [comment]
Could anyone affected by this collect logs? (run sway with -d and post the output of stdout) It seems there is an issue with direct scanout not bailing correctly.

**Comment by egnappahz** [comment]
Here are my repeating relevant errors when a freeze occurs:

```
00:00:13.454 [ERROR] [wlr] [backend/drm/renderer.c:88] Failed to import source buffer into multi-GPU renderer
00:00:13.454 [ERROR] [sway/desktop/output.c:317] Page-flip failed on output HDMI-A-2
00:00:13.454 [ERROR] [wlr] [backend/drm/renderer.c:88] Failed to import source buffer into multi-GPU renderer
00:00:13.454 [ERROR] [sway/desktop/output.c:317] Page-flip failed on output HDMI-A-2
00:00:13.455 [ERROR] [wlr] [backend/drm/renderer.c:88] Failed to import source buffer into multi-GPU renderer
00:00:13.455 [ERROR] [sway/desktop/output.c:317] Page-flip failed on output HDMI-A-2
00:00:13.455 [ERROR] [wlr] [backend/drm/renderer.c:88] Failed to import source buffer into multi-GPU renderer
00:00:13.455 [ERROR] [sway/desktop/output.c:317] Page-flip failed on output HDMI-A-2
...
```

**Comment by 532910** [comment]
```
00:00:24.628 [DEBUG] [sway/commands/bind.c:618] running command for binding: fullscreen global
00:00:24.628 [INFO] [sway/commands.c:261] Handling command 'fullscreen global'
00:00:24.628 [DEBUG] [sway/desktop/transaction.c:798] Transaction 0x55b8f136ef10 committing with 1 instructions
00:00:24.728 [DEBUG] [sway/desktop/transaction.c:884] Transaction 0x55b8f136ef10 is ready
00:00:24.728 [DEBUG] [sway/desktop/transaction.c:694] Applying transaction 0x55b8f136ef10
00:00:24.729 [DEBUG] [sway/input/cursor.c:888] denying request to set cursor from unfocused client
```

**Comment by 532910** [comment]
this issue is about `fullscreen global` not about `fullscreen`
please open another issues for `fullscreen` without `global`

**Comment by egnappahz** [comment]
I seriously doubt that.

**Comment by egnappahz** [comment]
For me adding the variable

```
WLR_SCENE_DISABLE_DIRECT_SCANOUT=1
```

was a good enough workaround, I see what you mean now @Nefsen402

**Comment by 532910** [comment]
`WLR_SCENE_DISABLE_DIRECT_SCANOUT=1` changes nothing
I have no errors in the debug log
please stop mixing different issued into a single one

**Comment by egnappahz** [comment]
You are right, you have another problem.
But freezes were mentioned here. Left it out there for people with a similar problem so they can have a better understanding what is what, because maybe everything is crystal clear to you, it was not to me.

**Comment by emersion** [comment]
Can you post a full debug log please? Not just a snippet?

**Comment by emersion** [comment]
@egnappahz, your issue is tracked in https://github.com/swaywm/sway/issues/8404

**Comment by 532910** [comment]
I found what's going on! It so happene that I only have one monitor now, and when commenting out parts of the sway config I left a not entirely correct entry:
```
output DP-1position 2160,0 scale 2
```

Though I still believe this is an issue, as
1. previous sway version (1.9-2) worked fine with this config.
2. I see nothing at all, black screen, while `output DP-1position 100,100 scale 2` gives more expected result the global fullscreen mode: a window moved 100 pixels up and 100 pixels left out of the visible area.

**Comment by jerrymattias** [comment]
We also see issues with the fullscreen global in sway 1.10. It works well in 1.9.  The regression we see is that it is not possible to set the display positions, i.e side by side.  We also do not see the expected mirrored screen if both displays has the same positions. 

While we normally use Chromium as the application this is also reproducable with i.e the foot terminal.
 To reproduce 

-Connect two displays.
-Start foot.
-Enable global fullscreen use swaymsg (not config file) 
-Positions all screens to 0 0 use swaymsg (not config file) 

Expected here is that the applications should mirror on both screens. I only see it on one screen. 

From all screens at 0 0
-Positions one screen to 1920 0 use swaymsg (not config file) 
Application will expand to both screens. (It looks right at first glance)
-Positions same screen back to 0 0 use swaymsg (not config file) 
-Positions the other screen to 1920 0 use swaymsg (not config file) 
Expected here would be that the outputs should have changed place on the screens but they show up in the same order as before.

**Comment by emersion** [comment]
https://github.com/swaywm/sway/pull/8609 should help.
